src/csv.ts

```diff
@@ -23,34 +25,52 @@ type EventType = `${Event}`;
 
 type ColumnHeader = { name: string; type: DataType };
 
-export class CSV {
+export class CSV<D extends string> {
     protected _options: CsvLoaderOptions;
     protected _loader: Loader;
-    protected _handlers: Map<Event, Set<EventHandler>>;
+    protected _handlers: Map<EventType, Map<D, Set<EventHandler>>>;
+
+    #openedDataSource: D;
```

Comment:
I would like to use this opportunity for a discussion regarding accessibility modifiers and property names.

Is there a reason for using `protected` by default?
I personally use the approach of being as restricted as possible, i.e., if it isn't part of the public interface and isn't needed by an inheriting class, then it should be private by default.

Another thing is the underscore prefix for variable names.
I am aware of this being an idiom/convention made up by JS developers to mark class members as private (although they are not private at runtime). Depending on who you ask it is considered as good or bad practice.
Alternatively, this prefix is often used to mark unused variables in order to ignore them for linters. This is done in Deno, Rust, and even a proposed option for [ESLint](https://eslint.org/docs/rules/no-unused-vars#argsignorepattern).
When it comes to prefix private members, I use the new [private class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) feature, which is currently a stage 3 proposal in [ECMAScript](https://github.com/tc39/proposal-class-fields) (and already implemented in all modern browsers). This enables using private class members in JS as well, meaning they are private and cannot be accessed at runtime.

Reply:
On the discussion on whether `private` or `protected` should be the default: I prefer making as much of a class as possible accessible for inheritance. I do not want to assume which parts of a class should be customizable and which not, as I can not claim to know every possible use case.
If you go out of your way to inherit a existing class instead of just using it because the original author decided to not support or did not think of your use case, it should not be unnecessarily hard to change the behavior. At this point, you're' out of the original implementation's scope anyway, so it should not restrict you in adapting the class.
My stance on this developed due to multiple cases of having to deal with unnecessarily strict inheritance interfaces, which made re-use impossible and resulted in avoidable work being required, either adapting the interface or even re-implementing the library's functionality.

As for the usage of the proper JS private syntax: I'm glad this feature is finally coming to ECMAScript, but I'd prefer to delay the adoption until the 2022 standard is officially released.

Reply:
I understand your point of view and can agree on that. I will adapt the private accessors to comply to the existing protected accessors.